tests/time_tests/test_runner/conftest.py

```diff
@@ -17,36 +17,63 @@
 
 # pylint:disable=import-error
 import pytest
+import argparse
 from pathlib import Path
 import yaml
+import hashlib
 
-from test_runner.utils import expand_env_vars
+from test_runner.utils import expand_env_vars, upload_timetest_data, \
+    DATABASE, DB_COLLECTIONS
 
 
 # -------------------- CLI options --------------------
 
+
 def pytest_addoption(parser):
     """Specify command-line options for all plugins"""
-    parser.addoption(
+    test_args_parser = parser.getgroup("test run")
+    test_args_parser.addoption(
         "--test_conf",
         type=Path,
-        help="Path to test config",
+        help="path to test config",
         default=Path(__file__).parent / "test_config.yml"
     )
-    parser.addoption(
+    test_args_parser.addoption(
         "--exe",
         required=True,
         dest="executable",
-        type=Path,
-        help="Path to a timetest binary to execute",
+        type=str,
+        help="path to a timetest binary to execute"
     )
-    parser.addoption(
+    test_args_parser.addoption(
         "--niter",
         type=int,
-        help="Number of iterations to run executable and aggregate results",
+        help="number of iterations to run executable and aggregate results",
         default=3
     )
     # TODO: add support of --mo, --omz etc. required for OMZ support
+    db_args_parser = parser.getgroup("test database reporting")
+    db_args_parser.addoption(
+        '--db_report',
+        nargs=3,
+        type=str,
+        metavar=("ID", "URL", "COLLECTION"),
+        help='enable reporting to database:' \
+             ' ID - unique identifier for preparing unique record,' \
+             ' URL - MongoDB URL in "mongodb://server:port" format,' \
+             ' COLLECTION - collection name in "{}" database'.format(DATABASE),
```

Comment:
URL and COLLECTION should be separate args to align with other tools:
    parser.add_argument('--db_url', required=not is_dryrun,
                        help='MongoDB URL in a for "mongodb://server:port".')
    parser.add_argument('--db_collection', required=not is_dryrun,
                        help=f'Collection name in {DATABASE} database to upload.',
                        choices=DB_COLLECTIONS)

It is better to keep DB configuration options separated from the actions tool doing. That will allow better manageability of the configuration in infrastructure runs those scripts.

Reply:
Done